packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -45,73 +67,82 @@ contract RepPriceOracle is IRepPriceOracle, Initializable {
         return _newExchangeData.price;
     }
 
-    function getRepPriceInAttoCash(IV2ReputationToken _reputationToken) external view returns (uint256) {
-        return calculateNewExchangeData(_reputationToken).price;
-    }
-
-    function calculateNewExchangeData(IV2ReputationToken _reputationToken) private view returns (ExchangeData memory) {
+    function calculateNewExchangeData(IV2ReputationToken _reputationToken) private returns (ExchangeData memory) {
         ExchangeData memory _exchangeData = exchangeData[address(_reputationToken)];
         uint256 _blockNumber = block.number;
+        uint256 _blockTimestamp = block.timestamp; // solium-disable-line security/no-block-members
         if (_blockNumber == _exchangeData.blockNumber) {
             return _exchangeData;
         }
 
+        if (uniswapUpgraded && !_exchangeData.upgradeHandled) {
+            uint256 _price = _exchangeData.price;
+            initializeUniverse(_reputationToken);
+            exchangeData[address(_reputationToken)].price = _price;
+            exchangeData[address(_reputationToken)].upgradeHandled = true;
+            return exchangeData[address(_reputationToken)];
+        }
```

Comment:
Once again trying _really hard_ to ignore the upgradable exchange while reviewing...

It feels like this code should be extracted out into `pokeRepPriceInAttoCash` if possible.  That would allow us to mark this function as `view` I believe, which is valuable for auditing purposes (isolate view code from non-view code).

Reply:
the exchange.sync makes this not `view`-able